compiler/src/main/java/dagger/internal/codegen/BindingGraph.java

```diff
@@ -954,8 +954,9 @@ void resolve(BindingKey bindingKey) {
           /* Resolve in the parent in case there are multibinding contributions or conflicts in some
            * component between this one and the previously-resolved one. */
           parentResolver.get().resolve(bindingKey);
-          if (!new LocalDependencyChecker().dependsOnLocalBindings(bindingKey)
-              && getLocalExplicitBindings(bindingKey.key()).isEmpty()) {
+          LocalDependencyChecker localDependencyChecker = new LocalDependencyChecker();
+          if (!localDependencyChecker.dependsOnLocalBindings(bindingKey) &&
+              !localDependencyChecker.bindingDefinedLocally(bindingKey)) {
```

Comment:
I think we want to leave this as it currently is in the PR.

If we start checking if a binding is defined locally and hence conflicts with a previously resolved binding in `localDependencyChecker.dependsOnLocalBindings()` this makes determining the owning component of a binding in `Resolver.getOwningComponent()` more challenging. Checking if a binding is defined locally in a separate method that can be reused in other places makes things more composable and straightforward. 

@netdpb @ronshapiro Does that seems reasonable or is there some reason why we'd want to check for binding conflicts in `.dependsOnLocalBindings()`?